hack/docgen.sh

```diff
@@ -3,12 +3,12 @@ set -euo pipefail
 
 compatibilitymatrix() {
     go run hack/docs/version_compatibility.go hack/docs/compatibility-karpenter.yaml "$(git describe --exact-match --tags || echo "no tag")"
-    go run hack/docs/compatibilitymetrix_gen_docs.go website/content/en/preview/upgrading/compatibility.md hack/docs/compatibility-karpenter.yaml 6
+    go run hack/docs/compatibilitymetrix_gen_docs.go website/content/en/preview/upgrading/compatibility.md hack/docs/compatibility-karpenter.yaml 7
```

Comment:
musing: What does this number mean again? Would be nice to use CLI args so it's a little easier to reason about

Reply:
The number of supported k8s versions. This makes sure we still include 1.23 rather than starting at 1.24.

Reply:
Can at least make the value a variable to give use some context?

Reply:
Sure, that's easy enough. Probably should update all of the hack programs to use named flags at some point. I'm imagining just `--file` and `--version-count` here.